### PR TITLE
Add update endpoints for aws s3 storage folder + sagemaker notebook

### PR DESCRIPTION
**Pull request by rogerwangcs**
For s3 storage folders and sagemaker notebook resources:
- Add update endpoints to swagger
- Add update controller method that calls update resource flights with just common update parameters (name, description)
- Remove error throw from s3 folder and sagemaker resource `updateSteps`
- leaving `updateSteps` todos up if/when we need to add cloud resource update steps.

Since these endpoints only call the shared update resource flight without any additional steps, how would we test?

Tested locally:
PATCH `/api/workspaces/v1/[workspaceId]/resources/controlled/aws/storageFolder/[resourceid]`
```
request body:
{
	"name": "new-aws-storage-folder",
	"description": "testing description"
}

response body:
{
	"metadata": {
		"workspaceId": "0d9d305a-bb09-4da0-a3d6-95d0d448a810",
		"resourceId": "4560983c-685e-4abc-9270-cd7c13fcb977",
		"name": "new-aws-storage-folder",
		"description": "testing description",
		"resourceType": "AWS_S3_STORAGE_FOLDER",
		"stewardshipType": "CONTROLLED",
		"cloudPlatform": "AWS",
		"cloningInstructions": "COPY_NOTHING",
		"controlledResourceMetadata": {
			"accessScope": "SHARED_ACCESS",
			"managedBy": "USER",
			"privateResourceUser": {},
			"privateResourceState": "NOT_APPLICABLE",
			"region": "us-east-1"
		},
		"resourceLineage": [],
		"properties": [],
		"createdBy": "rogerwangcs@google.com",
		"createdDate": "2023-06-07T21:06:07.669941Z",
		"lastUpdatedBy": "rogerwangcs@google.com",
		"lastUpdatedDate": "2023-06-08T12:52:43.66046Z",
		"state": "READY"
	},
	"attributes": {
		"bucketName": "v0-saas-devel-us-east-1-terra",
		"prefix": "test-update-storage-folder-0"
	}
}
```

PATCH `/api/workspaces/v1/[workspaceId]/resources/controlled/aws/storageFolder/[resourceid]`
```
request body:
{
	"name": "new-sagemaker-notebook-name",
	"description": "testing"
}

response body:
{
	"metadata": {
		"workspaceId": "0d9d305a-bb09-4da0-a3d6-95d0d448a810",
		"resourceId": "d14c9bc8-cf1e-4963-b602-fc748e9462ca",
		"name": "new-sagemaker-notebook-name",
		"description": "testing",
		"resourceType": "AWS_SAGEMAKER_NOTEBOOK",
		"stewardshipType": "CONTROLLED",
		"cloudPlatform": "AWS",
		"cloningInstructions": "COPY_NOTHING",
		"controlledResourceMetadata": {
			"accessScope": "PRIVATE_ACCESS",
			"managedBy": "USER",
			"privateResourceUser": {
				"userName": "rogerwangcs@google.com"
			},
			"privateResourceState": "ACTIVE",
			"region": "us-east-1"
		},
		"resourceLineage": [],
		"properties": [],
		"createdBy": "rogerwangcs@google.com",
		"createdDate": "2023-06-08T13:43:22.921959Z",
		"lastUpdatedBy": "rogerwangcs@google.com",
		"lastUpdatedDate": "2023-06-08T14:07:37.64493Z",
		"state": "READY"
	},
	"attributes": {
		"instanceName": "test-sagemaker-nb-update",
		"instanceType": "ml.t2.medium"
	}
}
```

### CODE REVIEW

File: openapi/src/parts/controlled_aws_sagemaker_notebook.yaml [hunk 1]
```diff
@@ -88,6 +88,25 @@ paths:
...
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/UpdateControlledAwsSageMakerNotebookRequestBody'
+      responses:
+        '200':
+          $ref: '#/components/responses/UpdateControlledAwsSageMakerNotebookResponse'
+        '403':
```
Comment (dexamundsen) [inline openapi/src/parts/controlled_aws_sagemaker_notebook.yaml:104]: 409 may be needed

File: openapi/src/parts/controlled_aws_sagemaker_notebook.yaml [hunk 2]
```diff
@@ -216,3 +246,10 @@ components:
         application/json:
           schema:
             $ref: '#/components/schemas/AwsSageMakerNotebookResource'
+
+    UpdateControlledAwsSageMakerNotebookResponse:
+      description: Response for an AWS SageMaker Notebook update.
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/AwsSageMakerNotebookResource'
```
Comment (dexamundsen) [inline openapi/src/parts/controlled_aws_sagemaker_notebook.yaml:255]: nit: new line

File: service/src/main/java/bio/terra/workspace/app/controller/ControlledAwsResourceApiController.java [hunk 3]
```diff
@@ -77,6 +81,8 @@ public class ControlledAwsResourceApiController extends ControlledResourceContro
 
   private final Logger logger = LoggerFactory.getLogger(ControlledAwsResourceApiController.class);
 
+  private final WsmResourceService wsmResourceService;
```
Comment (dexamundsen) [inline service/src/main/java/bio/terra/workspace/app/controller/ControlledAwsResourceApiController.java:84]: consider moving WsmResourceService to ControlledResourceControllerBase

File: openapi/src/parts/controlled_aws_s3_storage_folder.yaml [hunk 4]
```diff
@@ -62,6 +62,25 @@ paths:
...
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/UpdateControlledAwsS3StorageFolderRequestBody'
+      responses:
+        '200':
+          $ref: '#/components/responses/UpdateControlledAwsS3StorageFolderResponse'
+        '403':
```
[the remark this answers is not in the document]
Comment (ddietterich) [reply openapi/src/parts/controlled_aws_s3_storage_folder.yaml:78]: I think will for the case where you try to update a resource that is still being created or is being deleted or is being updated by someone else.

File: service/src/main/java/bio/terra/workspace/app/controller/ControlledAwsResourceApiController.java (hunk 3 above)
Comment (ddietterich) [reply service/src/main/java/bio/terra/workspace/app/controller/ControlledAwsResourceApiController.java:84]: I want to go the opposite direction. I think everything in ControlledResourceControllerBase should be in a "utils" module. There isn't a need for a base class.

File: service/src/main/java/bio/terra/workspace/app/controller/ControlledAwsResourceApiController.java [hunk 5]
```diff
@@ -316,6 +324,41 @@ public ResponseEntity<ApiAwsS3StorageFolderResource> getAwsS3StorageFolder(
...
+  public ResponseEntity<ApiAwsS3StorageFolderResource> updateAwsS3StorageFolder(
+      UUID workspaceUuid,
+      UUID resourceUuid,
+      @Valid ApiUpdateControlledAwsS3StorageFolderRequestBody body) {
+    logger.info(
+        "Updating AWS S3 Storage Folder resourceId {} workspaceUuid {}",
+        resourceUuid,
+        workspaceUuid);
+    AuthenticatedUserRequest userRequest = getAuthenticatedInfo();
+    workspaceService.validateWorkspaceAndContextState(workspaceUuid, CloudPlatform.AWS);
```
Comment (ddietterich) [inline service/src/main/java/bio/terra/workspace/app/controller/ControlledAwsResourceApiController.java:338]: Thanks for picking up the new pattern!

**Review by ddietterich — APPROVED** [review]

File: openapi/src/parts/controlled_aws_s3_storage_folder.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (rogerwangcs) [reply openapi/src/parts/controlled_aws_s3_storage_folder.yaml:78]: I added the 409, but it doesn't cover the case of resource name conflicts. But it looks like its an issue for all resources at the moment. I created a ticket to track: https://broadworkbench.atlassian.net/browse/PF-2831.

**Pull request merged**